dbt/include/spark/macros/materializations/snapshot.sql

```diff
@@ -33,10 +38,18 @@
 {% macro spark_build_snapshot_staging_table(strategy, sql, target_relation) %}
     {% set tmp_identifier = target_relation.identifier ~ '__dbt_tmp' %}
 
-    {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
-                                                  schema=target_relation.schema,
-                                                  database=none,
-                                                  type='view') -%}
+    {% if target_relation.is_iceberg %}
+      {# iceberg catalog does not support create view, but regular spark does. We removed the catalog and schema #}
+      {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
+                                                    schema=none,
+                                                    database=none,
+                                                    type='view') -%}
+    {% else %}
+      {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
+                                                    schema=target_relation.schema,
```

Comment:
Since `target_relation` is aware that it's an iceberg relation, can we update target_relation.schema to be None in that scenario? Then this piece of code doesn't care whether or not it's an iceberg relation.

Reply:
Can you elaborate a bit @mikealfare. I'm not a fan of mutating the `target_relation` since it is being used at multiple places. We only want to do this for the `tmp_relation`.

Reply:
This logic can be removed as soon as the view implementation is ready. The spec is already finalized and can be found [here](https://iceberg.apache.org/view-spec/).